frontend2/src/utils/api.ts

```diff
@@ -297,31 +300,33 @@ export const getAllUserTournamentSubmissions = async (
  * Create a new user.
  * @param user The user's info.
  */
-export const createUser = async (
-  user: customModels.CreateUserInput
-): Promise<models.UserCreate> => {
-  const defaultUser = {
+export const createUser = async ({
+  username,
+  password,
+  email,
+  firstName,
+  lastName,
+  profile: { gender, genderDetails, school, country },
+}: customModels.CreateUserInput): Promise<models.UserCreate> => {
+  const userCreate: models.UserCreate = {
     id: -1,
-    isStaff: false,
+    is_staff: false,
+    username,
+    password,
+    email,
+    first_name: firstName,
+    last_name: lastName,
     profile: {
-      avatarUrl: "",
-      hasAvatar: false,
-      hasResume: false,
+      gender: gender as models.GenderEnum,
+      country,
+      gender_details: genderDetails,
+      avatar_url: "",
+      has_avatar: false,
+      has_resume: false,
     },
   };
   // convert our input into models.UserCreate.
-  return (
-    await API.apiUserUCreate({
-      ...defaultUser,
-      ...user,
-      profile: {
-        ...defaultUser.profile,
-        ...user.profile,
-        gender: user.profile.gender as unknown as models.GenderEnum,
-        country: user.profile.country as unknown as models.CountryEnum,
-      },
-    })
-  ).body;
+  return await API.apiUserUCreate({ userCreate });
```

Comment:
Just want to confirm that this new generated api function call (await foo() vs. await foo().body) returns the same model as the old one so that no breaking bugs pop up in the already-built UI 😁

Reply:
yup, tht's correct